src/pages/page.js

```diff
@@ -47,4 +47,10 @@ export default class Page {
    * @returns {Element|null}
    */
   findInsertionPoint(widgetZone) {}
+
+  /**
+   * Returns the name of the CustomElement of the widget button.
+   * If not defined, `obe-{platform}-button` will be used.
+   */
+  getButtonElementName() {}
```

Comment:
❤️ Lovely change for making this even more extensible

Reply:
Wondered about that! Currently `Page` has no idea of the platform it’s on as far as I can tell, so we’d need to add a variable or something to those classes. And if we do, I’m wondering if it’d be better to use `page.platform` rather than passing in `platform` as a param in `addWidgetElements`. What do you think?

Reply:
Pros:
- platform is stored in a more logical place than as a floating string in the entrypoint files
- lets us implement this function

Cons:
- means adding state to the page, which until now exists as an interface
- duplicated string for different pages for the same platform

I think I'm in favour of adding this to the page. But worry about this impacting future expansion, or being a blocker for adding future sites. But really can't think of any reason why it would be 🤔 WDYT?

Reply:
I gave it a stab in https://github.com/orbit-love/orbit-browser-extension/pull/51/commits/694c587642282ea3693eff4166bdf3257d85cc23. Let me know if you like it—I ended up deciding for a method (`getPlatform()`) rather than state, which as you point out was a bit of an odd duck.

Reply:
💜 Love it. Think the function is a great call, but could be worth adding a comment about it in the JSDocs (*which are not correct atm, heads up) since it's a bit weird without the context of use trying to avoid state here. Happy to go ahead with that

Reply:
Yep, great call! Updated in [Update comment](https://github.com/orbit-love/orbit-browser-extension/pull/51/commits/491a7427f05d02440e913fa27f78bf032f0799fa)